mne/viz/evoked.py

```diff
@@ -1198,3 +1199,398 @@ def plot_evoked_joint(evoked, times="peaks", title='', picks=None,
+    truncate_yaxis : bool
+        If True, the left y axis is truncated to half the max value and
+        rounded to .25 to reduce visual clutter.
+    invert_y : bool
+        Controls wheter negative values are plotted up (as is sometimes done
+        for ERPs out of tradition). Defaults to False.
+    ax : None | `matplotlib.pyplot.axes` instance
+        What axes to plot to. If None, a new axes is created.
+    ymin, ymax : float | None
+        If not `None`, can be used to manually scale the y axis.
+    title : None | str
+        If str, will be plotted as figure title. If None, the channel
+        names will be shown.
+    show : bool
+        If True, show the figure.
+
+    Returns
+    -------
+    fig : Figure
+        The figure in which the plot is drawn.
+    """
+    import matplotlib.pyplot as plt
+    from ..evoked import Evoked
+
+    # set up labels and instances
+    if isinstance(evokeds, Evoked):
+        conditions = ['Evoked']  # empty title
+        evokeds = dict(Evoked=evokeds)
+    else:
+        if isinstance(evokeds, dict):
+            conditions = sorted(list(evokeds.keys()))
+        else:
+            evokeds = dict((str(ii + 1), evoked)
+                           for ii, evoked in enumerate(evokeds))
+            conditions = sorted(list(evokeds.keys()))
+
+    # get and set a few limits and variables (times, channels, units)
+    example = (evokeds[conditions[0]]
+               if isinstance(evokeds[conditions[0]], Evoked)
+               else evokeds[conditions[0]][0])
+    if not isinstance(example, Evoked):
+            raise ValueError("evokeds must be an `mne.Evoked` "
+                             "or of a collection of `mne.Evoked`s")
+    times = example.times
+    tmin, tmax = times[0], times[-1]
+
+    # deal with picks: infer indices and names
+    if picks == 'gfp':
+        if ymin is None:
+            ymin = 0
+        gfp_picks = pick_types(example.info, eeg=True, meg=True, seeg=True,
+                               ecog=True)
+        ch_types = list(set([channel_type(example.info, pick)
+                             for pick in gfp_picks]))
+        if len(ch_types) > 1:
+            raise ValueError("Can't plot GFP with multiple channel types.")
+        ch_type = ch_types[0]
+        ch_names = ['Global Field Power']
+    else:
+        if isinstance(picks, int):
+            picks = [picks]
+        elif isinstance(picks[0], int):
+            pass
+        else:
+            raise ValueError("`picks` must be int, a list of int, "
+                             "or `gfp`, not " + str(type(picks)))
+        ch_names = [example.ch_names[pick] for pick in picks]
+        ch_type = channel_type(example.info, picks[0])
+        ch_types = list(set(channel_type(example.info, pick_)
+                        for pick_ in picks))
+        if len(ch_types) > 1:
+            raise ValueError("More than 1 channel type selected by `picks`.")
```

Comment:
You do this separately for gfp. Could it be factorized to do it for both cases at the same time?